### PR TITLE
Introduced config schema for plugins/catalog-backend

### PR DESCRIPTION
**Pull request by keshann93**
## Hey, I just made a Pull Request!

Ref [#3364](https://github.com/backstage/backstage/issues/3364)

I have raised the PR including config schema for catalog-backend

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by benjdlambert** [comment]
@freben could you give this a check plz.

File: plugins/catalog-backend/config.d.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      ldapOrg?: {
+        providers: {
+          target: string;
+          bind?: { [key: string]: string };
+          users: { [key: string]: string | { [key: string]: string } };
+          groups: { [key: string]: string | { [key: string]: string } };
+        };
+      };
+      microsoftGraphOrg?: {
+        providers: { [key: string]: string };
```
Comment (freben) [inline plugins/catalog-backend/config.d.ts:39]: https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/ingestion/processors/microsoftGraph/config.ts#L19

File: plugins/catalog-backend/config.d.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      allow: Array<string>;
+    }>;
+    processors?: {
+      githubOrg?: {
+        providers: Array<{
+          [key: string]: string;
+        }>;
+      };
+      ldapOrg?: {
+        providers: {
```
Comment (freben) [inline plugins/catalog-backend/config.d.ts:31]: https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/ingestion/processors/ldap/config.ts#L25

File: plugins/catalog-backend/config.d.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          bind?: { [key: string]: string };
+          users: { [key: string]: string | { [key: string]: string } };
+          groups: { [key: string]: string | { [key: string]: string } };
+        };
+      };
+      microsoftGraphOrg?: {
+        providers: { [key: string]: string };
+      };
+    };
+    locations?: Array<{ [key: string]: string }>;
```
Comment (freben) [inline plugins/catalog-backend/config.d.ts:42]: this is an array of `{ key: string, value: string }` objects

File: plugins/catalog-backend/config.d.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  /** Configuration options for the catalog plugin */
+
+  catalog?: {
+    rules?: Array<{
```
Comment (freben) [inline plugins/catalog-backend/config.d.ts:21]: https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/ingestion/CatalogRules.ts#L40

**Review by freben — COMMENTED** [review]
I think we have to make these a lot more precise. I have linked to a bunch of places where the actual model is defined, and I would like to have both the structure and all of the comments added, ideally. This is essentially tech documentation for people, so it's super useful that it's clear and correct. Let me know if that's OK with you.

File: plugins/catalog-backend/config.d.ts (hunk 4 above)
Comment (keshann93) [reply plugins/catalog-backend/config.d.ts:21]: Thank you @freben! will do the needful!

**Comment by keshann93** [comment]
@freben Thank you for the review, I have updated the schema based on the comments, let me know whether current schema definition details are acceptable or further needs to be drill down

**Review by freben — APPROVED** [review]
Thanks! I have some ideas for tweaks but they may be out of scope for the PR itself - I'll merge and tweak.

**Pull request merged**